torchgeo/datasets/mapinwild.py

```diff
@@ -0,0 +1,328 @@
+                   "https://huggingface.co/datasets/burakekim/mapinwild/resolve/main/s2_spring/s2_spring_part2.zip"},
+    "s2_summer" : {"https://huggingface.co/datasets/burakekim/mapinwild/resolve/main/s2_summer/s2_summer_part1.zip",
+                    "https://huggingface.co/datasets/burakekim/mapinwild/resolve/main/s2_summer/s2_summer_part2.zip"},
+     "s2_winter" : {"https://huggingface.co/datasets/burakekim/mapinwild/resolve/main/s2_winter/s2_winter_part1.zip",
+                    "https://huggingface.co/datasets/burakekim/mapinwild/resolve/main/s2_winter/s2_winter_part2.zip"}
+    }
+    
+    split_url = "https://huggingface.co/datasets/burakekim/mapinwild/resolve/main/split_IDs/split_IDs.csv"
+    main_directory = os.path.join(os.getcwd(), "data")
+    mask_palette = {1 : (0,153,0),
+                0 : (255,255,255)} 
+
+    def __init__(
+        self,
+        root: str = "data",
+        modality: list = ["mask","esa_wc","s2_temporal_subset","viirs","s1"],
+        split: str = "train",
+        transforms: Optional[Callable[[Dict[str, Tensor]], Dict[str, Tensor]]] = None,
+        download: bool = False,
+        checksum: bool = False
+    ) -> None:
+        """Initialize a new MapInWild dataset instance.
+        Args:
+            root: root directory where dataset can be found
+            modality: the modality to download
+            split: one of "train", "val", or "test"
+            transforms: a function/transform that takes input sample and its target as
+                entry and returns a transformed version
+            download: if True, download dataset and store it in the root directory
+            checksum: if True, check the MD5 of the downloaded files (may be slow)
+            
+        Raises:
+            AssertionError: if ``split`` argument is invalid
+        """
+        assert split in ["train", "validation", "test"]
+
+        self.checksum = checksum
+        self.root = root
+        self.transforms = transforms
+        self.modality = modality
+        self.modality.remove("mask")
+
+        self.download = download
+        self._verify_split() 
+
+        split_dataframe = pd.read_csv(os.path.join(self.root, "split_IDs.csv"))
+        self.ids = split_dataframe[split].dropna().values.tolist() 
+        self.ids = [int(i) for i in self.ids]
+        
+        #Check if the requested list of modalities exist in the directory  
+        if not set(self.modality).issubset(set(os.listdir(self.main_directory))):
+            for modal in modality: 
+                for modality_link in self.modality_urls[modal]:
+                    self._verify(modality_link)
+                
+            for modal in modality:
+                if len(self.modality_urls[modal]) > 1:
+                    self.merge_parts(self.main_directory, modal)
+
+        self.list_modals = []
+        
+    def __getitem__(self, index: int) -> Dict[str, Tensor]:
+        """Return an index within the dataset.
+        Args:
```

Comment:
Missing spaces between sections, this happens when you copy-n-paste code from GitHub